components/module.py

```diff
@@ -0,0 +1,27 @@
+class Module:
+    # instances
+    # module_number = 0
+
+    # **kwargs for json
+    def __init__(self, module_number, **kwargs):
+        self.module_number = module_number
+        module_number += 1
```

Comment:
why are we incrementing it by 1?

Reply:
originally i was going to set the module_number as a global var and just increase it every time after a module is created, but i guess we can implement that later

Reply:
good practice is to avoid use of global variables -- they are too difficult to predict in python. The idea was each module object would maintain its own module number.